qa-tests-backend/src/main/groovy/services/NotifierService.groovy

```diff
@@ -19,13 +21,16 @@ class NotifierService extends BaseService {
         return NotifierServiceGrpc.newBlockingStub(getChannel())
     }
 
+    @Retry
     static addNotifier(NotifierOuterClass.Notifier notifier) {
         return getNotifierClient().postNotifier(notifier)
     }
 
     static testNotifier(NotifierOuterClass.Notifier notifier) {
         try {
-            getNotifierClient().testNotifier(notifier)
+            Helpers.withRetry(3, 1) {
+                getNotifierClient().testNotifier(notifier)
+            }
```

Comment:
Does it really need to be wrapped with `try..catch`?
I would expect the test framework to catch the exception, log the error and fail the test.

Reply:
That's what we try to avoid. If there is a network issue we don't want to fail the test until we retry 3 times

Reply:
How does `try..catch` help with that?

Reply:
try catch is used to keep same interface return false on error and true on success

Reply:
This is a helper method not a test. If we remove catch here we will change interface and need to rewrite test that is using it:
https://github.com/stackrox/stackrox/blob/842c86654941f0e25d242d746193abf7754db839/qa-tests-backend/src/test/groovy/IntegrationsTest.groovy#L85

I'm also not a fan of this approach but in this case having a boolean indicating if it's ok or not is way better than what we have in other places where we swallow exception and return null or empty string